src/main/java/build/buildfarm/worker/CASFileCache.java

```diff
@@ -347,12 +346,7 @@ public Path put(Digest digest, boolean isExecutable, Digest containingDirectory)
   }
 
   private static void setPermissions(Path path, boolean isExecutable) throws IOException {
-    ImmutableSet.Builder<PosixFilePermission> perms = new ImmutableSet.Builder<PosixFilePermission>()
-      .add(PosixFilePermission.OWNER_READ);
-    if (isExecutable) {
-      perms.add(PosixFilePermission.OWNER_EXECUTE);
-    }
-    Files.setPosixFilePermissions(path, perms.build());
+    new File(path.toString()).setExecutable(false, isExecutable);
```

Comment:
@excitoon this looks like this breaks linux? this change doesn't seem to set executable permissions on files extracted from the cache at all. I think this should look like `.setExecutable(isExecutable, true)`

Reply:
Yeah, you're right. Somehow I thought first argument is for other users and second is for owner.